src/ray/gcs/format/gcs.fbs

```diff
@@ -349,3 +349,14 @@ table ActorCheckpointIdData {
   // A list of the timestamps for each of the above `checkpoint_ids`.
   timestamps: [long];
 }
+
+// This enum type is used as object's metadata to indicate the object's creating
+// task has failed because of a certain error.
+enum ErrorType:int {
+  // Indicates that a task failed because the worker died.
+  WORKER_DIED = 1,
+  // Indicates that a task failed because the actor died.
+  ACTOR_DIED = 2,
```

Comment:
It would be great to also differentiate between intentional actor failures vs unexpected failures. Another useful error might be failure due to node death. I don't think we need to do these now, but can you add them as `TODO`s?

Reply:
sounds good